pkg/api/persistentvolume/util.go

```diff
@@ -25,14 +25,19 @@ import (
 // DropDisabledFields removes disabled fields from the pv spec.
 // This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a pv spec.
 func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.PersistentVolumeSpec) {
-	if !utilfeature.DefaultFeatureGate.Enabled(features.ExpandCSIVolumes) && !hasExpansionSecrets(oldPVSpec) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ExpandCSIVolumes) {
 		if pvSpec.CSI != nil {
-			pvSpec.CSI.ControllerExpandSecretRef = nil
+			if !hasControllerExpansionSecrets(oldPVSpec) {
+				pvSpec.CSI.ControllerExpandSecretRef = nil
+			}
+			if !hasNodeExpansionSecrets(oldPVSpec) {
+				pvSpec.CSI.NodeExpandSecretRef = nil
```

Comment:
This field check should be toggled by CSINodeExpandSecret feature gate

Reply:
I had remove this change, I think it's not needed.